src/main/java/collab/DataBase.java

```diff
@@ -2,84 +2,92 @@
 
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.HashSet;
 import java.util.List;
 import java.util.stream.Collectors;
-import collab.Employee;
 
 public class DataBase {
-  private ArrayList<Employee> employeeData = new ArrayList<Employee>();
+  HashSet<String> registerEmployeeNum = new HashSet<String>();
+  public ArrayList<Employee> employeeData = new ArrayList<Employee>();
   
   public ArrayList<Employee> getEmployeeAllData() {
     return employeeData;
   }
   
   public void add(Employee employee) {
-    employeeData.add(employee);        
+    if(registerEmployeeNum.contains(employee.getEmployeeNumber())) {
+      System.out.println(employee.getEmployeeNumber() + "Register Fail!! Employee Data Duplicate!!");
+      return;
+    }
+    registerEmployeeNum.add(employee.getEmployeeNumber());
+    employeeData.add(employee);
   }
-
+  
+  
   public Employee searchItem(String id) {
-    return searchItems("employeeNum", id).get(0);
+    List<Employee> foundItems = searchItems("employeeNum", id); 
+    if(foundItems.size()==0) return null;
+    return foundItems.get(0);
   }
 
   public List<Employee> searchItems(String field, String value) {
-      List<Employee> resultItem = new ArrayList<Employee>();
-      resultItem.addAll(employeeData.stream()
+      return employeeData.stream()
               .filter(item -> item.getField(field).equals(value))
-              .collect(Collectors.toList()));
-      return resultItem;
+              .collect(Collectors.toList());
   }
   
   public Employee modifyItem(Employee employee) {
-    Employee foundItem = searchItems("employeeNum", employee.getEmployeeNumber()).get(0);
+    Employee foundItem = searchItem(employee.getEmployeeNumber());
+    if(foundItem == null) return null;
+    
     Employee returnItem = makeEmployee(foundItem);
     foundItem = employee;
-    
     return returnItem;
   }
   
   public Employee modifyItemById(String id, String field, String value) {
-    Employee foundItem = modifyItemByCondition("employeeNum", id, field, value).get(0);
-    return foundItem;
+    List<Employee> foundItems = modifyItemByCondition("employeeNum", id, field, value);
+    if(foundItems.size()==0) return null;
```

Comment:
이 부분도 위와 마찬가지로 Colleciton이라서 다음과 같은 코드가 어떨까요?
SonarCloud 추천 코드 내용입니다.

```suggestion
    if(foundItems.isEmpty()) return null;
```